src/models.zig

```diff
@@ -153,6 +165,193 @@ pub const Model = struct {
+				var coordsIter = std.mem.split(u8, line[3..], " ");
+				var uv: Vec2f = undefined;
+				var i: usize = 0;
+				while (coordsIter.next()) |coord| : (i += 1) {
+					uv[i] = std.fmt.parseFloat(f32, coord) catch |e| blk: { std.log.err("Failed parsing {s} into float: {any}", .{coord, e}); break :blk 0; };
+				}
+				uv[0] *= 4;
+				uv[1] *= 4;
+				uvs.append(.{uv[0], uv[1]});
+			} else if (std.mem.eql(u8, line[0..2], "f ")) {
+				const n = std.mem.count(u8, line[2..], " ") + 1;
+				switch (n) {
+					3 => {
+						var coordsIter = std.mem.split(u8, line[2..], " ");
+						var faceData: [3][3]usize = undefined;
+						var i: usize = 0;
+						var failed = false;
+						while (coordsIter.next()) |vertex| : (i += 1) {
+							var d = std.mem.split(u8, vertex, "/");
+							var j: usize = 0;
+							if (std.mem.count(u8, vertex, "/") != 2 or std.mem.count(u8, vertex, "//") != 0) {
+								failed = true;
+								std.log.err("Failed loading face {s}. Each vertex must use vertex/uv/normal", .{line, n});
+								break;
+							}
+							while (d.next()) |value| : (j += 1) {
+								faceData[j][i] = std.fmt.parseUnsigned(usize, value, 10) catch |e| blk: { std.log.err("Failed parsing {s} into uint: {any}", .{value, e}); break :blk 1; };
+								faceData[j][i] -= 1;
+							}
+						}
+						if (!failed)
+							tris.append(.{.vertex=faceData[0], .uvs=faceData[1], .normals=faceData[2]});
+					},
+					4 => {
+						var coordsIter = std.mem.split(u8, line[2..], " ");
+						var faceData: [3][4]usize = undefined;
+						var i: usize = 0;
+						var failed = false;
+						while (coordsIter.next()) |vertex| : (i += 1) {
+							var d = std.mem.split(u8, vertex, "/");
+							var j: usize = 0;
+							if (std.mem.count(u8, vertex, "/") != 2 or std.mem.count(u8, vertex, "//") != 0) {
+								failed = true;
+								std.log.err("Failed loading face {s}. Each vertex must use vertex/uv/normal", .{line, n});
+								break;
+							}
+							while (d.next()) |value| : (j += 1) {
+								faceData[j][i] = std.fmt.parseUnsigned(usize, value, 10) catch |e| blk: { std.log.err("Failed parsing {s} into uint: {any}", .{value, e}); break :blk 1; };
+								faceData[j][i] -= 1;
+							}
+						}
+						if (!failed)
+							quadFaces.append(.{.vertex=faceData[0], .uvs=faceData[1], .normals=faceData[2]});
+					},
+					else => std.log.err("Failed loading face {s} with {d} vertices", .{line, n})
+				}
+			}
+		}
+
+		var quadInfos = main.List(QuadInfo).init(main.stackAllocator);
+		defer quadInfos.deinit();
+
+		for (tris.items) |face| {
+			const normal: Vec3f = normals.items[face.normals[0]];
+			
+			var uvA: Vec2f = uvs.items[face.uvs[0]];
+			var uvB: Vec2f = uvs.items[face.uvs[2]];
+			var uvC: Vec2f = uvs.items[face.uvs[1]];
+
+			const minUv = @floor(@min(@min(uvA, uvB), uvC));
+
+			if (minUv[0] < 0 or minUv[0] > 4 or minUv[1] < 0 or minUv[1] > 4) {
+				std.log.err("Uv value for model is outside of 0-1 range", .{});
```

Comment:
I think this case is actually useful when you want to repeat the texture inside the model.
So I would just remove this check.

Reply:
this check is here so that the textureSlot doesnt become like -1 or 17 because the uv is outside of the 0-1 range

Reply:
Oh, I see.